tests/datacon/pos/AutoliftedFields01.hs

```diff
@@ -0,0 +1,16 @@
+{-@ LIQUID "--exact-data-cons" @-}
+
+-- data decl in LH and Haskell give different names to the fields, but use them
+-- in valid ways.
+
+module AutoliftedFields01 where
+
+{-@ type Nat = { v : Int | v >= 0 } @-}
+type Nat = Int
+
+{-@ data T = T { getMyT :: Nat } @-}
+data T = T { getT :: Nat }
```

Comment:
I would have expected naming a field differently in the specification to be an error. Shouldn't it fail?